### PR TITLE
debug: Add a document to introduce serial log enable for celadon

### PR DESCRIPTION
**Pull request by btian1**
Tracked-On: OAM-84511
Signed-off-by: Tian Baofeng <baofeng.tian@intel.com>

### CODE REVIEW

File: source/blogs/uart/uart.rst [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+On WHL,  HW board has a UART with 9-pins connector output. We just need connect it to PC host with a serial2USB converter cable.
+
+.. figure:: images/usb2uart_female.jpg
+    :align: center
+
+SW support
+~~~~~~~~~~
+
+enable kernel log serial in mixins.spec for the project before image building.
```
Comment (ttzeng) [inline source/blogs/uart/uart.rst]: Suggest to rewrite "Modify the ``serialport`` setting in the *mixins.spec* configuration file, and re-build the Celadon image."

File: source/blogs/uart/uart.rst [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    .. code-block:: text
+
+	serialport: ttyS0
+
+On PC host, open the ttyUSB device to capture logs
```
Comment (ttzeng) [inline source/blogs/uart/uart.rst]: Suggest to rewrite "On the host PC, you can open the serial port to capture the debug logs with the following command:"

File: source/blogs/uart/uart.rst [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    .. code-block:: bash
+
+	sudo minicom -D /dev/ttyUSB0
+
+Status
+~~~~~~
+
+The logs from BIOS,Kernelflinger, Kernel and AOSP can be output during startup.
```
Comment (ttzeng) [inline source/blogs/uart/uart.rst]: Suggest to rewrite "You will get the output logs from the BIOS, Kernelflonger, Linux kernel, and Celadon Android stack during the board starting up."

File: source/blogs/uart/uart.rst [hunk 4]
```diff
@@ -0,0 +1,118 @@
+Celadon Serial debug logs output enable
+=======================================
+
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! thanks!

File: source/blogs/uart/uart.rst [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+HW connection
+~~~~~~~~~~~~~
+
+On KBL_NUC, HW board has a UART output, extend it out with the connector below:
+
+.. figure:: images/kbl_uart_connector.jpg
+    :align: center
+
+Connect to KBL_NUC:
+
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! thanks!

File: source/blogs/uart/uart.rst [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+.. figure:: images/kbl_extend_uart.jpg
+    :align: center
+
+Then connect it to PC host with a serial2USB converter cable.
+
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    .. code-block:: text
+
+	serialport: ttyUSB0
+
+On PC host, open the ttyUSB device to capture logs
+
+    .. code-block:: bash
+
+	sudo minicom -D /dev/ttyUSB0
+Because Kernelflinger has no ttyUSB driver, the logs is saved to a file.
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+Table of Contents
+#################
+
+.. contents::
+    :depth: 2
+    :local:
+
+Serial debug logs output enable on KBL_NUC
+------------------------------------------
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+	sudo minicom -D /dev/ttyUSB0
+
+Status
+~~~~~~
+
+The logs from BIOS,Kernelflinger, Kernel and AOSP can be output during startup.
+
+Serial debug logs output enable on APL_NUC
+------------------------------------------
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    :depth: 2
+    :local:
+
+Serial debug logs output enable on KBL_NUC
+------------------------------------------
+
+HW connection
+~~~~~~~~~~~~~
+
+On KBL_NUC, HW board has a UART output, extend it out with the connector below:
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+Then connect it to PC host with a serial2USB converter cable.
+
+.. figure:: images/usb2uart_female.jpg
+    :align: center
+
+SW support
+~~~~~~~~~~
+
+enable kernel log serial in mixins.spec for the project before image building.
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+    .. code-block:: text
+
+	serialport: ttyS0
+
+On PC host, open the ttyUSB device to capture logs
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+
+    .. code-block:: bash
+
+	sudo minicom -D /dev/ttyUSB0
+
+Status
+~~~~~~
+
+The logs from BIOS,Kernelflinger, Kernel and AOSP can be output during startup.
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+
+Serial debug logs output enable on APL_NUC
+------------------------------------------
+
+HW connection
+~~~~~~~~~~~~~
+
+On APL_NUC, HW board has no UART output.For kernel and AOSP logs output, we need a cable below to translate USB to serial, ttyUSB driver is supported in kernel:
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+
+.. figure:: images/usb2uart_male.jpg
+    :align: center
+
+Then connect it to PC host with a serial2USB converter cable.
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+
+.. figure:: images/usb2uart_female.jpg
+    :align: center
+
+SW support
+~~~~~~~~~~
+
+Enable kernel log serial in mixins.spec for the project before image building.
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+
+    .. code-block:: text
+
+	serialport: ttyUSB0
+
+On PC host, open the ttyUSB device to capture logs
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+
+    .. code-block:: bash
+
+	sudo minicom -D /dev/ttyUSB0
+Because Kernelflinger has no ttyUSB driver, the logs is saved to a file.
+
+Status
+~~~~~~
+
+The logs from Kernelflinger is saved to a file. And Kernel and AOSP logs can be captured in a console, e.g.: minicom.
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Add the log file location and rewrite the status description. 
Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 19]
```diff
@@ -0,0 +1,118 @@
...
+
+Serial debug logs output enable on whisklake (WHL)
+--------------------------------------------------
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst [hunk 20]
```diff
@@ -0,0 +1,118 @@
...
+
+HW connection
+~~~~~~~~~~~~~
+
+On WHL,  HW board has a UART with 9-pins connector output. We just need connect it to PC host with a serial2USB converter cable.
```
[the remark this answers is not in the document]
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst (hunk 1 above)
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst (hunk 2 above)
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

File: source/blogs/uart/uart.rst (hunk 3 above)
Comment (btian1) [reply source/blogs/uart/uart.rst]: Done! Thanks!

**Comment by ttzeng** [comment]
> @ttzeng , please review again.

@btian1 there are two warnings during the Sphinx compilation - line 12 and 53. You have to extend the length of the section header characters to match the length of the section titles. The rest of document looks good. Thanks.

**Pull request merged**